plugins/out_stackdriver/stackdriver.c

```diff
@@ -277,6 +277,145 @@ static char *get_google_token(struct flb_stackdriver *ctx)
     return ctx->o->access_token;
 }
 
+static int process_local_resource_id(const void *data, size_t bytes,
+                                     struct flb_stackdriver *ctx, char *type)
+{
+    int i;
+    int len;
+    int ret = 0;
+    int flag = 0;
+    size_t off = 0;
+    msgpack_object k;
+    msgpack_object v;
+    msgpack_object root;
+    msgpack_object_map map;
+    msgpack_unpacked result;
+    char *local_resource_id;
+    char *ptr;
+    const char delim[] = ".";
+
+    msgpack_unpacked_init(&result);
+    while (msgpack_unpack_next(&result, data, bytes, &off) == MSGPACK_UNPACK_SUCCESS && !flag) {
+        root = result.data;
+        flag = 0;
+
+        if (root.type != MSGPACK_OBJECT_ARRAY ||
+            root.via.array.size != 2 ||
+            root.via.array.ptr[1].type != MSGPACK_OBJECT_MAP) {
+            flb_plg_warn(ctx->ins, "unexpected record format");
+            continue;
+        }
+
+        map = root.via.array.ptr[1].via.map;
+        len = strlen(LOCAL_RESOURCE_ID_KEY);
+        for (i = 0; i < map.size; i++) {
+            k = map.ptr[i].key;
+            v = map.ptr[i].val;
+
+            if (k.type == MSGPACK_OBJECT_STR && 
+                strncmp(k.via.str.ptr, LOCAL_RESOURCE_ID_KEY, len) == 0) {
+
+                local_resource_id = malloc(v.via.str.size+1);
+                strncpy(local_resource_id, v.via.str.ptr, v.via.str.size);
+                local_resource_id[v.via.str.size] = '\0';
+
+                ptr = strtok(local_resource_id, delim);
+
+                if (strncmp(type, "k8s_container", strlen("k8s_container")) == 0) {
+                    /* check the prefix */
+                    if (strncasecmp(ptr, type, strlen("k8s_container")) != 0) {
+                        ret = -1;
+                    }
+
+                    ptr = strtok(NULL, delim);
+                    /* the local_resource_id for k8s_container is in format: 
```

Comment:
I think the comment is not obvious here. I still suggest to put it in stackdriver_format function before call local_resource_id. Like: https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/master/lib/fluent/plugin/out_google_cloud.rb#L1131

Reply:
updated